### PR TITLE
Fix the graph track content top margin

### PR DESCRIPTION
**Pull request by vickyliu-go4it**
In graph tracks, the top context margin was count twice as it is inserted
both above and below the legends. But this should only happen when there
are legends.

Bug: http://b/218517055

### CODE REVIEW

File: src/OrbitGl/GraphTrack.cpp [hunk 1]
```diff
@@ -37,24 +37,27 @@ GraphTrack<Dimension>::GraphTrack(CaptureViewElement* parent,
...
 template <size_t Dimension>
 float GraphTrack<Dimension>::GetLegendHeight() const {
-  if (IsCollapsed() || Dimension <= 1) {
-    return 0;
-  }
   // This is a bit of an arbitrary choice - I don't want to introduce an additional layout
   // parameter just for the legend size, but it should be smaller than all regular textboxes
   // across Orbit.
-  return layout_->GetTextBoxHeight() / 2.f;
+  return HasLegend() ? layout_->GetTextBoxHeight() / 2.f : 0;
```
Comment (karupayun) [inline src/OrbitGl/GraphTrack.cpp:60]: I didn't see this before. If you want to introduce a new variable, consider in the future to add a method in Layout which simply returns GetTextBoxHeight()/2. In this way all the logic respect of sizes is in this class.

File: src/OrbitGl/GraphTrack.cpp [hunk 2]
```diff
@@ -37,24 +37,27 @@ GraphTrack<Dimension>::GraphTrack(CaptureViewElement* parent,
...
+}
+
 template <size_t Dimension>
 float GraphTrack<Dimension>::GetHeight() const {
-  // Top content margin is counted twice because it it inserted before and after the legend
-  float height = layout_->GetTrackTabHeight() + layout_->GetTrackContentTopMargin() * 2 +
-                 GetLegendHeight() + GetGraphContentHeight() +
-                 layout_->GetTrackContentBottomMargin();
-  return height;
+  // Top content margin is counted twice when there are legends because it is inserted above and
```
Comment (karupayun) [inline src/OrbitGl/GraphTrack.cpp:47]: The content margin MIGHT be counted twice....

File: src/OrbitGl/GraphTrack.cpp [hunk 3]
```diff
@@ -37,24 +37,27 @@ GraphTrack<Dimension>::GraphTrack(CaptureViewElement* parent,
...
+  // below the legend.
+  float track_content_top_margin = (HasLegend() ? 2.f : 1.f) * layout_->GetTrackContentTopMargin();
```
Comment (karupayun) [inline src/OrbitGl/GraphTrack.cpp:49]: I would rather include the LegendHeight here.

**Review by karupayun — APPROVED** [review]
LGTM! Thanks for solving this! :) I left a few minor comments.

File: src/OrbitGl/GraphTrack.cpp (hunk 2 above)
Comment (vickyliu-go4it) [reply src/OrbitGl/GraphTrack.cpp:47]: No might here. When there are legends, it MUST be counted twice.

File: src/OrbitGl/GraphTrack.cpp (hunk 1 above)
Comment (vickyliu-go4it) [reply src/OrbitGl/GraphTrack.cpp:60]: I guess this comments might be added while Florian was improving this part.

I'm kind of agree with the comment here. The legend is only used in graph tracks, I prefer to not add and additional layout parameter.

Comment (karupayun) [reply src/OrbitGl/GraphTrack.cpp:60]: Sorry, I was not clear. If you DON'T want to introduce ... 

You can simply add a method without a parameter.

File: src/OrbitGl/GraphTrack.cpp (hunk 2 above)
Comment (karupayun) [reply src/OrbitGl/GraphTrack.cpp:47]: When there are legends yes, but not always. Again, I would prefer to use "might".

Comment (karupayun) [reply src/OrbitGl/GraphTrack.cpp:47]: Wow, I've just seen it. Please ignore my comment

File: src/OrbitGl/GraphTrack.cpp [hunk 4]
```diff
@@ -37,24 +37,29 @@ GraphTrack<Dimension>::GraphTrack(CaptureViewElement* parent,
...
 template <size_t Dimension>
 float GraphTrack<Dimension>::GetHeight() const {
-  // Top content margin is counted twice because it it inserted before and after the legend
-  float height = layout_->GetTrackTabHeight() + layout_->GetTrackContentTopMargin() * 2 +
-                 GetLegendHeight() + GetGraphContentHeight() +
-                 layout_->GetTrackContentBottomMargin();
-  return height;
+  // Top content margin is counted twice when there are legends because it is inserted above and
+  // below the legend.
+  float legend_height_with_margins =
```
Comment (karupayun) [inline src/OrbitGl/GraphTrack.cpp:49]: I don't like this name. 

I would rather prefer: "legend_height_and_margins" and include bottom_margin, or something like "height_above_content" as you suggested in the chat a few days ago.

Comment (vickyliu-go4it) [reply src/OrbitGl/GraphTrack.cpp:49]: We can have an offline discussion on the naming if you don't mind

**Pull request merged**